CMakeLists.txt

```diff
@@ -51,6 +51,14 @@ endif()
 set(NVIM_VERSION_MAJOR 0)
 set(NVIM_VERSION_MINOR 1)
 set(NVIM_VERSION_PATCH 1)
+# Package maintainers should define this according to their package system
+# conventions, if the source code or runtime files were modified, e.g.:
+#   cmake -DNVIM_VERSION_PRERELEASE=-deb3
+# or define it as empty to prevent "-dev", if no modifications were made:
+#   cmake -DNVIM_VERSION_PRERELEASE=
+if(NOT DEFINED NVIM_VERSION_PRERELEASE)
+  set(NVIM_VERSION_PRERELEASE "-dev")
+endif()
```

Comment:
> AFAIK that is how this PR functions. In which case would that not be the case (except in a git tree)? 

You can build any revision with `-DNVIM_VERSION_PRERELEASE=`, which would make it look like a release. We hardcode the version number, so I feel like we should also hardcode this part of the version and make it do "the right thing" out of the box. It's more of a personal preference thing, though.

> How about other projects (e.g., Rust, Atom--I did not check these myself)?

I will check some more projects.

> Section 9 of semver.org says arbitrary pre-release data may be appended after a hyphen...

Yes, but git describe appends "post-release data", i.e. it's supposed to be 0.1.0 < 0.1.0-1, whereas semver would treat it as 0.1.0-1 < 0.1.0, as the pre-release data after the hyphen is a string, not a number (see also https://github.com/mojombo/semver/issues/106). Also, if we do the 0.1.0 -> 0.1.1-dev versioning scheme, then we would have a version of e.g. 0.1.1-dev for tarball builds and 0.1.0-5-abcdef for `git describe` builds.. maybe we should not use git describe directly, but e.g. `0.1.1-dev` for tarball builds and `0.1.1-dev+abcdef` for Git tree builds? :grimacing:

If we decide to change anything, I would create the PR, so you don't have to do all the work.. I hope I'm not overcomplicating things :grimacing: